hub/src/server/tlsServer.ts

```diff
@@ -0,0 +1,97 @@
+import { Application as ExpressApp } from 'express'
+import * as https from 'https'
+import * as fs from 'fs'
+import { TlsCertConfigInterface, TlsPfxCert, TlsPemCert } from './config'
+
+function tryDecodeBase64(content: string): Buffer | false {
+  try {
+    const buffer = Buffer.from(content, 'base64')
+    if (buffer.toString('base64') === content) {
+      return buffer
+    } else {
+      return false
+    }
+  } catch (error) {
+    return false
+  }
+}
+
+const PEM_PREFIX = '-----BEGIN '
+
+function isTlsPfxCert(tlsConfig: TlsCertConfigInterface): tlsConfig is TlsPfxCert {
+  return !!(tlsConfig as TlsPfxCert).pfxFile
+}
+
+function isTlsPemCert(tlsConfig: TlsCertConfigInterface): tlsConfig is TlsPemCert {
+  return !!(tlsConfig as TlsPemCert).keyFile
+}
+
+export function loadPemCert(tlsConfig: TlsPemCert, opts: https.ServerOptions) {
+  if (tlsConfig.keyFile.startsWith(PEM_PREFIX)) {
```

Comment:
Can we guarantee that `.keyFile` will be stripped of whitespace?

Reply:
Fixed